sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/TableServiceClient.java

```diff
@@ -430,9 +630,88 @@ public TableServiceProperties getProperties() {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<TableServiceProperties> getPropertiesWithResponse(Duration timeout, Context context) {
-        return blockWithOptionalTimeout(client.getPropertiesWithResponse(context), timeout);
+        ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+        long timeoutInMillis = setTimeout(timeout);
+        Future<Response<TableServiceProperties>> future = scheduler.submit(() -> getPropertiesWithResponse(context));
+        scheduler.shutdown();
+        try {
+            return future.get(timeoutInMillis, TimeUnit.MILLISECONDS);
+        } catch (Exception ex) {
+            throw logger.logExceptionAsError((RuntimeException) TableUtils.mapThrowableToTableServiceException(ex));
+        }
+    }
+
+    Response<TableServiceProperties> getPropertiesWithResponse(Context context) {
+        context = context == null ? Context.NONE : context;
+        Response<com.azure.data.tables.implementation.models.TableServiceProperties> response =
+            this.implementation.getServices().getPropertiesWithResponse(null, null, context);
+        return new SimpleResponse<>(response, toTableServiceProperties(response.getValue()));
+    }
+
+    private TableServiceProperties toTableServiceProperties(
```

Comment:
This can be shared with the Async client, consider moving it to commons.
Applicable here and below.

Reply:
[Ditto question here](https://github.com/Azure/azure-sdk-for-java/pull/33143#discussion_r1084439563)